api/cluster/base.go

```diff
@@ -0,0 +1,127 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"net/http"
+
+	"github.com/banzaicloud/pipeline/pkg/cluster"
+	pkgCommon "github.com/banzaicloud/pipeline/pkg/common"
+	"github.com/banzaicloud/pipeline/secret"
+)
+
+// CreateClusterRequestBase describes the common base of cluster creation requests
+type CreateClusterRequestBase struct {
+	Name         string               `json:"name" yaml:"name" binding:"required"`
+	Features     []Feature            `json:"features"`
+	SecretID     string               `json:"secretId" yaml:"secretId"`
+	SecretName   string               `json:"secretName" yaml:"secretName"`
```

Comment:
Why do we need both secret name and secret ID?

Reply:
No idea. We had them in the old API, I thought it had significance. I'd gladly remove one of them.

Reply:
Because the user of the API might know either of them, and there is no API exposed to exchange them other than looking up in all secret's list. The mapping from name to ID is not defined from end user's perspective (and I hope it will change at some time).